lib/axon/compiler.ex

```diff
@@ -500,14 +525,26 @@ defmodule Axon.Compiler do
       Enum.map_reduce(
         inputs,
         cache_and_counts,
-        &to_model_funs(&1, nodes, &2, mode, key)
+        &to_model_funs(&1, nodes, &2, config)
       )
 
     # Names are computed lazily, so compute name from current
     # op and aggregate op_counts.
     name = name_fn.(op_name, op_counts)
     op_counts = Map.update(op_counts, op_name, 1, fn x -> x + 1 end)
 
+    # TODO: Hack for dropout with key, fix with a better implementation
```

Comment:
Perhaps it should not be based on the name but I would say it should be based on the key. Thoughts?